### PR TITLE
Add filter option for templatable runs

### PR DESCRIPTION
**Pull request by schustmi**
## Describe changes
Add filter option for templatable runs.

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/zen_stores/sql_zen_store.py [hunk 1]
```diff
@@ -974,7 +974,8 @@ def filter_and_paginate(
             RuntimeError: if the schema does not have a `to_model` method.
         """
         query = filter_model.apply_filter(query=query, table=table)
-
+        query = query.distinct()
```
Comment (bcdurak) [inline src/zenml/zen_stores/sql_zen_store.py:977]: Why did we have to add this?

File: src/zenml/models/v2/core/pipeline_run.py [hunk 2]
```diff
@@ -235,6 +235,9 @@ class PipelineRunResponseMetadata(WorkspaceScopedResponseMetadata):
         default=None,
         description="Template used for the pipeline run.",
     )
+    is_templatable: bool = Field(
```
Comment (bcdurak) [inline src/zenml/models/v2/core/pipeline_run.py:238]: This is probably intended, but I just wanted draw attention to the fact that this is a required field without a default. You might have to adjust some of the fixtures for our tests.

File: src/zenml/zen_stores/schemas/pipeline_run_schemas.py [hunk 3]
```diff
@@ -328,6 +328,14 @@ def to_model(
         )
         metadata = None
         if include_metadata:
+            is_templatable = False
```
Comment (bcdurak) [inline src/zenml/zen_stores/schemas/pipeline_run_schemas.py:331]: In the `get_custom_filters`, it seems like we are looking for pipeline builds that have `is_local` set to False. Should we include a similar logic here?

**Review by bcdurak — COMMENTED** [review]
Just left some initial comments.

File: src/zenml/models/v2/core/pipeline_run.py (hunk 2 above)
Comment (schustmi) [reply src/zenml/models/v2/core/pipeline_run.py:238]: Yep that's intended, let's see if our tests fail

File: src/zenml/zen_stores/schemas/pipeline_run_schemas.py (hunk 3 above)
Comment (schustmi) [reply src/zenml/zen_stores/schemas/pipeline_run_schemas.py:331]: Yep, added it here and also in one other place where we check something similar 👍

File: src/zenml/zen_stores/sql_zen_store.py (hunk 1 above)
Comment (bcdurak) [reply src/zenml/zen_stores/sql_zen_store.py:977]: It is certainly a nice idea. I was just curious what led to this case.😄

File: src/zenml/models/v2/core/pipeline_run.py (hunk 2 above)
Comment (bcdurak) [reply src/zenml/models/v2/core/pipeline_run.py:238]: There is one [here](https://github.com/zenml-io/zenml/blob/be215d157c27be99206429d3ca703c94ff46a542/tests/unit/conftest.py#L447).

**Review by bcdurak — APPROVED** [review]

**Pull request merged**